src/main/java/net/sf/jabref/importer/ParserResult.java

```diff
@@ -126,12 +126,8 @@ public boolean hasWarnings() {
         return !warnings.isEmpty();
     }
 
-    public String[] warnings() {
-        String[] s = new String[warnings.size()];
-        for (int i = 0; i < warnings.size(); i++) {
-            s[i] = warnings.get(i);
-        }
-        return s;
+    public ArrayList<String> warnings() {
+        return warnings;
```

Comment:
Please use `List` instead and create a new list if possible to not let a caller change the warnings from the outside.